src/spanningtrees/wilson.jl

```diff
@@ -0,0 +1,170 @@
+
+If you specify the root of the tree, the function produces a spanning tree that is picked uniformly among all trees rooted at r. 
+
+If you do not specify a root, the function produces a random tree from G, picking *uniformly* among all spanning trees of G (overall all possible roots). 
+
+It has two implementations, depending on whether G is directed or not. For undirected graphs, we pick a random root uniformly and call random_spanning_tree_with_root. For directed graphs, we use Alg. 2 from Wilson (1996), which generates random forests iteratively until a forest with a single tree is found. 
+
+In a directed graph, it is important to note that all branches in the tree need to point *towards* the root, so that there may not be a spanning tree with root r even if the graph is connected. 
+
+As an example, consider the graph 1 -> 2 -> 3. There is a spanning tree rooted at 3 (the graph itself), but no spanning tree rooted at 2 or 1. 
+
+In an undirected graph that is connected, a spanning tree always exists, wherever one roots it. 1 <-> 2 <-> 3, rooted at 2, for instance, has spanning tree 1->2<-3.
+
+
+# Arguments
+
+- G: a graph
+- optional: r, index of a node to serve as root
+- maxiter: interrupt the algorithm for directed graphs after maxiter attempts. Default 50. Ignored for undirected graphs.
+
+# Output 
+
+If the root is specified, returns the set of edges in the tree. If it isn't, returns a named tuple with 'edges': the set of edges and 'root': the root. 
+
+Interestingly, the root is a perfect sample from the stationary distribution of the random walk on the graph, see Wilson (1996). 
+
+
+# Examples
+
+The graph 1->2->3 has only one spanning tree (i.e. the graph itself)
+
+```jldoctest
+julia> G = PathDiGraph(3)
+{3, 2} directed simple Int64 graph
+
+julia> random_spanning_tree(G)
+(edges = LightGraphs.SimpleGraphs.SimpleEdge{Int64}[Edge 1 => 2, Edge 2 => 3], root = 3)
+```
+
+Other graphs have several:
+
+```@example
+julia> G = CycleDiGraph(4)
+{4, 4} directed simple Int64 graph
+
+julia> random_spanning_tree(G)
+(edges = LightGraphs.SimpleGraphs.SimpleEdge{Int64}[Edge 1 => 2, Edge 2 => 3, Edge 3 => 4], root = 4)
+
+julia> random_spanning_tree(G)
+(edges = LightGraphs.SimpleGraphs.SimpleEdge{Int64}[Edge 1 => 2, Edge 2 => 3, Edge 3 => 4, Edge 4 => 1], root = 1)
+
+julia> random_spanning_tree(G,2)
+3-element Array{LightGraphs.SimpleGraphs.SimpleEdge{Int64},1}:
+ Edge 1 => 2
+ Edge 3 => 4
+ Edge 4 => 1
+```
+
+# See also 
+
+prim_mst, kruskal_mst
+
+# References 
+
+Wilson, D. B. (1996). [Generating random spanning trees more quickly than the cover time](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.47.8598&rep=rep1&type=pdf). In STOC (Vol. 96, pp. 296-303).
+
+"""
+function random_spanning_tree end
+
+
+function random_spanning_tree(G :: AG ,r :: Integer) where AG <: AbstractGraph{T} where T
```

Comment:
It's not tight bounding, but it's good to apply a `,` or a `;` followed by a `space`. Like this `a, b; c`